snapcraft.yaml

```diff
@@ -342,6 +350,23 @@ parts:
       echo "ac_add_options --with-mozilla-api-keyfile=$SNAPCRAFT_STAGE/mozilla-desktop-geoloc-api.key" >> $MOZCONFIG
     override-prime: ''
 
+  debug-symbols:
+    plugin: nil
+    build-packages:
+      - python3
+      - python3-virtualenv
+    after:
+      - firefox
+    override-pull: |
+      if [ -f "$SNAPCRAFT_PROJECT_DIR/symbols-upload-token" ]; then
+        export SYMBOLS_ARCHIVE=$(find ../../firefox/build/obj-*/dist/ -type f -name "firefox-*.crashreporter-symbols.zip")
+        virtualenv venv/
+        source venv/bin/activate
+        venv/bin/pip3 install redo requests argparse
+        SOCORRO_SYMBOL_UPLOAD_URL=https://symbols.stage.mozaws.net/upload/ SOCORRO_SYMBOL_UPLOAD_TOKEN_FILE="$SNAPCRAFT_PROJECT_DIR/symbols-upload-token" venv/bin/python3 ../../firefox/src/toolkit/crashreporter/tools/upload_symbols.py $SYMBOLS_ARCHIVE
```

Comment:
This is for testing purpose still, I'm unsure how we would like to make this more flexible ?

Reply:
It looks like the `upload_symbols.py` script is standalone (other than the external dependencies that are being installed with pip3), so maybe have the firefox part copy it to `$SNAPCRAFT_STAGE` too, to avoid hardcoding a path again. Or maybe just fold this `debug-symbols` part into the `firefox` part?